### DISCUSSION TITLE
rotary and 48-pin artery

### DISCUSSION THREAD

**Discussion by pilou75** [keirf/flashfloppy#479] [Q&A]
Hi
does anybody could make it work ?
I don't have any other gotek or rotary to figure out what's wrong.

At first , I tried with the rotary and its HW-040 board (yes, the 3,3V pin was connected), I could use it (needs rotary=v2 option) BUT it always goes clockwise whatever I do (turn clockwise or counterclockwise). The select button was OK..
I double-check connections with multimeter from gotek PCB to rotary PCB, and to rotary.

Then I removed the PCB to connect directly the rotary, and it's worst, erratic behavior.

I wonder if something's wrong with the gotek, or rotary maybe.
Everything else works fine (OLED, buzzer)

firmware : 3.25
gotek SFRKC30AT3 / CBT7 chip
rotary : https://www.aliexpress.com/item/33039744482.html

**Comment by RetroClinic** [comment]
Hi.

Yes, they all work for me, but I don’t use cheapo encoders from Aliexpress, I use genuine Bourns:

https://uk.rs-online.com/web/p/mechanical-rotary-encoders/7377748/

Not sure if that’s your problem or not.

Mark.

From: pilou75 
Sent: Saturday, May 29, 2021 3:54 PM
To: keirf/FlashFloppy 
Cc: Subscribed 
Subject: [keirf/FlashFloppy] rotary and 48-pin artery 48 (#479)

Hi
does anybody could make it work ?
I don't have any other gotek or rotary to figure out what's wrong.

At first , I tried with the rotary and its HW-040 board (yes, the 3,3V pin was connected), I could use it (needs rotary=v2 option) BUT it always goes clockwise whatever I do (turn clockwise or counterclockwise). The select button was OK..
I double-check connections with multimeter from gotek PCB to rotary PCB, and to rotary.

Then I removed the PCB to connect directly the rotary, and it's worst, erratic behavior.

I wonder if something's wrong with the gotek, or rotary maybe.
Everything else works fine (OLED, buzzer)

firmware : 3.25
gotek SFRKC30AT3 / CBT7 chip
rotary : https://www.aliexpress.com/item/33039744482.html

**Reply by keirf** [reply]
Probably a rubbish encoder. Most cheap ones do work though.

**Comment by pilou75** [comment]
thanks for your (very) quick answers

i won't be able to change the rotary very soon, but i will let you if that's ok with the new one.

**Reply by Dietrich-L** [reply]
I can confirm, that my 48-PIn Artery works well under v3.25 with a cheapo encoder and oled display. The missing rotary header is not a problen, i soldered the pins for the progamming header anyway. All operations including formatting work with my 360 kB IMG discs, actually a 6502 system originally published by the european magazine ELEKTOR.

Cudo to Keir for a job very well done.

**Comment by pilou75** [comment]
Hi,

You were right.
I bought 2 new different encoders. 
One have better "feeling" but they both work. perfectly.

**Comment by pilou75** [comment]
Hi,
It's me again. The gotek works very well, it works so well that I bought a second one for an old PC.
The second board is a SFRKC30AT4.7, and I can't use the rotary encode. I have definitively no luck.
Switch is OK, but turn right/left doesn't do anything.

I have swapped the 2 rotary encoders I have (different models).
None of them work on the SFRKC30AT4.7
Both works fine on the SFRKC30AT3 (my first gotek)

So, I don't really understand where is the issue.
I have checked the pin header solders I have done, they looks great.
I have checked the 3.3V on the rotary encoder PCB -> OK

Any idea ?
![20210712_184059_1920x1080](https://user-images.githubusercontent.com/85026766/125331514-606f7600-e348-11eb-877f-c261f0a9342a.jpg)

**Reply by keirf** [reply]
This new board has the traditional rotary header. Use that.

**Reply by pilou75** [reply]
It's not better. 

I have also removed the pin header, cleaned what I did, just to make sure there were not shortcut, did it again.
I'm getting mad.

**Reply by keirf** [reply]
I'm away this week but I'm sure it must work as lots of people have this board now and no other people have reported trouble.

**Reply by pilou75** [reply]
Finally, it works...
I used the same usb stick for both gotek, so same FF.CFG
For some reasons, the AT3 board needs v2 option, while the AT4.7 needs full, v2 value makes it not functionnal.
Also, it has to be connected to the rotary header as you said, using the programming port is not suitable.

**Reply by keirf** [reply]
I can't explain that it may be to do with the specific rotary encoder.
